File: backend/mobile-leetnote-backend/src/main/java/com/example/leetnote_backend/model/DTO/ProblemDetailDTO.java
package com.example.leetnote_backend.model.DTO;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class ProblemDetailDTO {
    private Long id;
    private String title;
    String difficulty;
    String description;
    boolean isFavorite;
    boolean solved;
    private SolutionDTO solution;
}

File: frontend/mobile-leetnode-frontend/app/src/main/java/com/example/leetnote/data/model/ProblemDetailDTO.kt
package com.example.leetnote.data.model

import com.google.gson.annotations.SerializedName

data class SolutionDTO(
    val approach: String,
    val code: String,
    val timeComplexity: String,
    val spaceComplexity: String
)

data class ProblemDetailDTO (
    @SerializedName("problemId") val id: Long,
    val title: String,
    val description: String,
    val difficulty: String,
    val isFavorite: Boolean,
    val isSolved: Boolean,
    val solution: SolutionDTO
)